### PR TITLE
Added support for Brave Rewards UI on Android

### PR DESCRIPTION
**Pull request by emerick**
Fix https://github.com/brave/brave-browser/issues/6526

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by emerick** [comment]
Notes to reviewers: This is a direct port of the Rewards UI from `browser-android-tabs`, updated to account for build-related changes in `brave-core` and `brave-ui`. It doesn't incorporate the very latest content changes made to the Rewards UI since that time, however. Ads is currently disabled in the UI, but that's being handled in a separate issue.

File: components/brave_rewards/resources/ui/components/boxAlert/index.tsx [hunk 1]
```diff
@@ -130,7 +130,7 @@ export default class BoxAlert extends React.PureComponent<Props, State> {
                 </StyledMonthlyTips>
               </>
             : <StyledMessage>
-                {getLocale('adsNotSupported')}
+                {getLocale('adsNotSupportedRegion')}
```
Comment (petemill) [inline components/brave_rewards/resources/ui/components/boxAlert/index.tsx]: 👍

File: components/brave_rewards/resources/brave_rewards_static_resources.grd [hunk 2]
```diff
@@ -9,6 +9,7 @@
   <release seq="1">
     <includes>
       <!-- WebUI Brave Rewards Page resources -->
+      <include name="IDR_BRAVE_REWARDS_ANDROID_PAGE_HTML" file="android_page/brave_rewards_page.html" type="BINDATA" />
       <include name="IDR_BRAVE_REWARDS_PAGE_HTML" file="page/brave_rewards_page.html" type="BINDATA" />
       <include name="IDR_BRAVE_REWARDS_IMG_WALLET_BG" file="page/img/wallet_bg.svg" type="BINDATA" />
       <include name="IDR_BRAVE_REWARDS_IMG_CAPTCHA_BAT" file="page/img/captcha_bat.png" type="BINDATA" />
```
Comment (NejcZdovc) [inline components/brave_rewards/resources/brave_rewards_static_resources.grd]: this is not used afaik

File: components/brave_rewards/resources/android_page/reducers/wallet_reducer.ts [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+
+      state = {
+        ...state,
+        ui,
+        walletInfo
+      }
+      break
+    }
+    case types.ON_ADDRESSES: {
+      if (!action.payload.addresses) {
```
Comment (NejcZdovc) [inline components/brave_rewards/resources/android_page/reducers/wallet_reducer.ts]: this is not needed as we don't have add funds

File: components/brave_rewards/resources/android_page/qrUtils.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (NejcZdovc) [inline components/brave_rewards/resources/android_page/qrUtils.ts]: this is not needed as we don't have add funds

File: components/brave_rewards/resources/BUILD.gn [hunk 5]
```diff
@@ -8,9 +8,11 @@ pack_web_resources("page_generated_resources") {
   resource_name = "brave_rewards_page"
   output_dir = "$root_gen_dir/brave/components/brave_rewards/resources"
   # need webpack and GRD generation first
-  deps = [
-    "page"
-  ]
+  if (is_android) {
```
Comment (NejcZdovc) [inline components/brave_rewards/resources/BUILD.gn]: do we need new name?

File: components/brave_rewards/resources/brave_rewards_static_resources.grd (hunk 2 above)
Comment (emerick) [reply components/brave_rewards/resources/brave_rewards_static_resources.grd]: It's used here, we conditionally load the Android-specific page when running on Android:

https://github.com/brave/brave-core/pull/3886/files#diff-472c4e44690bc2b99a1d2d0905870c3cR1690

File: components/brave_rewards/resources/BUILD.gn (hunk 5 above)
Comment (emerick) [reply components/brave_rewards/resources/BUILD.gn]: Do you mean `android_page`? It's just the directory where the Android-specific pages are located. They were unfortunately different enough that I had to put them in a separate location.

File: components/brave_rewards/resources/android_page/reducers/wallet_reducer.ts (hunk 3 above)
Comment (emerick) [reply components/brave_rewards/resources/android_page/reducers/wallet_reducer.ts]: Removed.

File: components/brave_rewards/resources/android_page/qrUtils.ts (hunk 4 above)
Comment (emerick) [reply components/brave_rewards/resources/android_page/qrUtils.ts]: Removed.

File: components/brave_rewards/resources/brave_rewards_static_resources.grd (hunk 2 above)
Comment (NejcZdovc) [reply components/brave_rewards/resources/brave_rewards_static_resources.grd]: nvm, I missed that this is general one as I was not sure why we would need captcha on android

**Comment by emerick** [comment]
Build was aborted during install phase, but I've seen that run successfully before and not much has changed but rebases.

**Comment by bsclifton** [comment]
This is missing milestone- but was merged into 0.74.x. I believe this equates to 1.4? (0.71.x was 1.0)

**Comment by bsclifton** [comment]
Update: looks like it's 1.3 👍 
https://github.com/brave/brave-core/commit/535cdc34e4e1652587cd7092a9dae8a3fe030eba

**Comment by emerick** [comment]
@bsclifton Yes, 1.3 is correct. Sorry for the oversight!

**Pull request merged**